tests/test_monitor_system.py

```diff
@@ -0,0 +1,219 @@
+
+    mocked_virtual_memory = mocker.MagicMock()
+    mocked_virtual_memory.percent = 20
+    mocked_virtual_memory.total = 4 * 1024**3
+
+    mocked_disk_usage = mocker.MagicMock()
+    mocked_disk_usage.used = 16 * 1024**3
+    mocked_disk_usage.percent = 50
+    mocked_disk_usage.total = 32 * 1024**3
+
+    mocker.patch(
+        "dvclive.monitor_system.psutil.virtual_memory",
+        return_value=mocked_virtual_memory,
+    )
+    mocker.patch(
+        "dvclive.monitor_system.psutil.disk_usage",
+        side_effect=[
+            mocked_disk_usage,
+            OSError,
+            mocked_disk_usage,
+            mocked_disk_usage,
+        ],
+    )
+
+
+def test_monitor_system_is_false(tmp_dir, mocker):
+    mock_psutil(mocker)
+
+    cpu_monitor_mock = mocker.patch("dvclive.live.CPUMonitor")
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=False,
+    ) as live:
+        assert live.cpu_monitor is None
+
+    cpu_monitor_mock.assert_not_called()
+
+
+def test_monitor_system_is_true(tmp_dir, mocker):
+    mock_psutil(mocker)
+    cpu_monitor_mock = mocker.patch("dvclive.live.CPUMonitor", spec=CPUMonitor)
+
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=True,
+    ) as live:
+        cpu_monitor = live.cpu_monitor
+
+        assert isinstance(cpu_monitor, CPUMonitor)
+        cpu_monitor_mock.assert_called_once()
+
+        end_spy = mocker.spy(cpu_monitor, "end")
+        end_spy.assert_not_called()
+
+    # check the monitoring thread is stopped
+    end_spy.assert_called_once()
+
+
+def test_ignore_non_existent_directories(tmp_dir, mocker):
+    mock_psutil_with_oserror(mocker)
+
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=False,
+    ) as live:
+        non_existent_disk = "/non-existent"
+        monitor = CPUMonitor(
+            folders_to_monitor={"main": "/", "non-existent": non_existent_disk}
+        )
+        monitor(live)
+        metrics = monitor._get_metrics()
+        monitor.end()
+
+    assert not Path(non_existent_disk).exists()
+
+    assert f"{METRIC_DISK_USAGE_PERCENT}/non-existent" not in metrics
+    assert f"{METRIC_DISK_USAGE_GB}/non-existent" not in metrics
+    assert f"{METRIC_DISK_TOTAL_GB}/non-existent" not in metrics
+
+
+@pytest.mark.timeout(2)
+def test_monitor_system_metrics(tmp_dir, mocker):
+    mock_psutil(mocker)
+    with Live(
+        tmp_dir,
+        save_dvc_exp=False,
+        monitor_system=False,
+    ) as live:
+        live.cpu_monitor = CPUMonitor(interval=0.05, num_samples=4)
+        # wait for the metrics to be logged.
+        # METRIC_DISK_TOTAL_GB is the last metric to be logged.
+        while len(dpath.search(live.summary, METRIC_DISK_TOTAL_GB)) == 0:
+            time.sleep(0.001)
+        live.next_step()
+
+        _, latest = parse_metrics(live)
+
+    schema = {}
+    for name in [
```

Comment:
do we check for actual values in both tests?

Reply:
I change one test to check for values :)